core/includes/common.inc

```diff
@@ -8930,3 +8930,58 @@ function backdrop_get_filetransfer_info() {
   }
   return $info;
 }
+
+/**
+ * Loads a file.
+ *
+ * Examples:
+ * @code
+ *   // Load node.admin.inc from the Node module.
+ *   backdrop_load_file('module', 'node', 'node.admin.inc');
+ *
+ *   // Load template.php from the Basis theme.
+ *   backdrop_load_file('theme', 'basis', 'template.php');
+ * @endcode
+ *
+ * Do not use this function to load an install file. Instead use:
+ * @code
+ *   module_load_install()
+ * @endcode
+ * Do not use this function in a global context, since it requires Backdrop to
+ * be fully bootstrapped. Instead use:
+ * @code
+ *   require_once BACKDROP_ROOT . '/path/file'
+ * @endcode
+ *
+ * @param string $project_type
+ *   The type of project to load the file from (module, theme, layout, profile).
+ * @param string $project_name
+ *   The name of the project to load the file from (e.g. admin_bar).
+ * @param string $filename
+ *   The name of the file to load. Specify the full filename, including the
+ *   extension (e.g. admin_bar.theme.inc).
+ *
+ * @return string|bool
+ *   The full path to the loaded file, if successful. FALSE otherwise.
+ */
+function backdrop_load_file($project_type, $project_name, $filename) {
+  static $files = array();
+
+  if (isset($files[$project_name][$filename])) {
+    return $files[$project_name][$filename];
+  }
+
+  if (function_exists('backdrop_get_path')) {
+    $file = BACKDROP_ROOT . '/' . backdrop_get_path($project_type, $project_name) . "/$filename";
```

Comment:
Something about php micro-performance and parsing strings containing variables. This works better I've been told:

```suggestion
    $file = BACKDROP_ROOT . '/' . backdrop_get_path($project_type, $project_name) . '/' . $filename;
```